flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/TaskManagerRunnerTest.java

```diff
@@ -144,6 +148,58 @@ public void testGenerateTaskManagerResourceIDWithLocalRpcService() throws Except
 		assertThat(taskManagerResourceID.getResourceIdString(), containsString(InetAddress.getLocalHost().getHostName()));
 	}
 
+	@Test
+	public void testUnexpectedTaskManagerTerminationFailsRunnerFatally() throws Exception {
+		final CompletableFuture<Void> terminationFuture = new CompletableFuture<>();
+		final TestingTaskExecutorService taskExecutorService = TestingTaskExecutorService.newBuilder()
+				.setTerminationFuture(terminationFuture)
+				.build();
+		final TaskManagerRunner taskManagerRunner = createTaskManagerRunner(
+				createConfiguration(),
+				(configuration,
+				resourceID,
+				rpcService,
+				highAvailabilityServices,
+				heartbeatServices,
+				metricRegistry,
+				blobCacheService,
+				localCommunicationOnly,
+				externalResourceInfoProvider,
+				fatalErrorHandler) -> taskExecutorService);
```

Comment:
maybe introduce a utility method that returns this factory; it is easy to miss that the list of arguments is part of the factory, and not the constructor.

Reply:
Good point. I will update it.